force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -101,19 +105,21 @@ public with sharing class ClassesObjectsMethods {
      * @return The number of vowels in the string.
      */
     public static Integer countVowels(String str) {
-        // Initialize the result Integer
-
-        /*
-        for () {
-            // get individual characters from the string - hint use substring or split
-            // check if the character is a vowel
-            // if it is a vowel, increment the result Integer
+       
+        String slc = str.toLowerCase();
+        Integer result = 0;
+        for (Integer i = 0; i < slc.length(); i++) {
+             String c = str.substring(i, i + 1);
+             if (c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u') {
+                result += 1;
+             }
         }
-        */
-
-        return null; // Replace null with the variable you used to store the result
+        return result; 
```

Comment:
Try to use better variable names than slc or c. 

When you check for membership in a set—in this case, the set of vowels—I like to use a Set. I think it is much easier to reason about that conditional logic. 

```
 public static Integer countVowels(String str) {
        Integer vowelCount = 0;
        Set<String> vowels = new Set<String>{
                'a', 'e', 'i', 'o', 'u'
        };
        List<String> characters = str.split('');
        for (String character : characters) {
            if (vowels.contains(character.toLowerCase())) {
                vowelCount++;
            }
        }
        return vowelCount;
    }
 ```